core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/saved-workflow-section.component.ts

```diff
@@ -265,6 +290,39 @@ export class SavedWorkflowSectionComponent implements OnInit, OnChanges {
     this.router.navigate([`${ROUTER_WORKFLOW_CREATE_NEW_URL}`], { queryParams: { pid: this.pid } }).then(null);
   }
 
+  /**
+   * create a new workflow based on uploaded json file
+   */
+  public onClickUploadExistingWorkflowFromLocal = (file: NzUploadFile): boolean => {
+    var reader = new FileReader();
+    reader.readAsText(file as unknown as Blob);
+    reader.onload = () => {
+      try {
+        const result = reader.result;
+        if (typeof result !== "string") {
+          throw new Error("Incorrect format: file is not a string");
+        }
+        const workflowContent = JSON.parse(result).content as WorkflowContent;
+        const workflowName = JSON.parse(result).name as string;
+        this.workflowPersistService
+          .createWorkflow(workflowContent, workflowName)
+          .pipe(untilDestroyed(this))
+          .subscribe({
+            next: uploadedWorkflow => {
+              this.dashboardWorkflowEntries = [...this.dashboardWorkflowEntries, uploadedWorkflow];
+            },
+            error: (err: unknown) => alert(err),
+          });
+      } catch (error) {
+        this.notificationService.error(
+          "An error occurred when importing the workflow. Please import a workflow json file."
```

Comment:
@zuozhiw Another thought: since we have two ways of importing/exporting, and they have different formats, is it really a good idea to let them both have .json as the file handle? Would this confuse the user? Or maybe we can do some checking to warn the user in case a json file intended for main workspace is uploaded here.

Reply:
Since we are unifying the content format of the json file in PR to be the same as #1582, this problem does not exist anymore.